xbmc/cores/VideoPlayer/DVDMessageQueue.cpp

```diff
@@ -183,14 +182,15 @@ MsgQueueReturnCode CDVDMessageQueue::Get(CDVDMsg** pMsg, unsigned int iTimeoutIn
 
       if (item.message->IsType(CDVDMsg::DEMUXER_PACKET) && item.priority == 0)
       {
-        DemuxPacket* packet = static_cast<CDVDMsgDemuxerPacket*>(item.message)->GetPacket();
+        DemuxPacket* packet =
+            std::static_pointer_cast<CDVDMsgDemuxerPacket>(item.message)->GetPacket();
         if (packet)
         {
           m_iDataSize -= packet->iSize;
         }
       }
 
-      *pMsg = item.message->Acquire();
+      pMsg = item.message;
```

Comment:
`pMsg = std::move(item.message);` because it's popped right after, from what I can tell

Reply:
added, thanks